client/checkout/woopay/email-input-iframe.js

```diff
@@ -75,11 +75,17 @@ export const handleWooPayEmailInput = async (
 
 	//Checks if customer has clicked the back button to prevent auto redirect
 	const searchParams = new URLSearchParams( window.location.search );
+	const isSkipWoopayCookieSet = /skip_woopay=1/.test( document.cookie );
 	const customerClickedBackButton =
 		( typeof performance !== 'undefined' &&
 			performance.getEntriesByType( 'navigation' )[ 0 ].type ===
 				'back_forward' ) ||
-		searchParams.get( 'skip_woopay' ) === 'true';
+		searchParams.get( 'skip_woopay' ) === 'true' ||
+		isSkipWoopayCookieSet; // We enforce and extend the skipping to the entire user session.
+
+	if ( customerClickedBackButton && ! isSkipWoopayCookieSet ) {
+		document.cookie = 'skip_woopay=1; path=/';
```

Comment:
> [!NOTE]
> This is a thought to start a discussion, not a request for change.

Is there any reason to enforce HTTPS-only here via `;secure`? I don't _think_ so since there is no sensitive data in the cooke, but thought I'd throw the thought out there 🤔

Reply:
I don't see a reason to enforce `secure` because of what you pointed out yourself. It's not sensitive data and we don't manipulate that information in any way.